action/action-dsl/src/main/java/com/powsybl/action/dsl/afs/ActionScript.java

```diff
@@ -0,0 +1,111 @@
+import com.powsybl.contingency.Contingency;
+import com.powsybl.iidm.network.Network;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public class ActionScript extends ProjectFile implements StorableScript, ContingenciesProvider {
+
+    public static final String PSEUDO_CLASS = "actionScript";
+    public static final int VERSION = 0;
+
+    static final String SCRIPT_CONTENT = "scriptContent";
+
+    private static final FileIcon IAL_ICON = new FileIcon("ial", ActionScript.class.getResourceAsStream("/icons/ial16x16.png"));
+
+    private final List<ScriptListener> listeners = new ArrayList<>();
+
+    public ActionScript(ProjectFileCreationContext context) {
+        super(context, VERSION, IAL_ICON);
+        storage.addListener(this, eventList -> {
+            for (NodeEvent event : eventList.getEvents()) {
+                if (event.getType() == NodeEventType.NODE_DATA_UPDATED) {
+                    NodeDataUpdated dataUpdated = (NodeDataUpdated) event;
+                    if (dataUpdated.getId().equals(info.getId()) && SCRIPT_CONTENT.equals(dataUpdated.getDataName())) {
+                        for (ScriptListener listener : listeners) {
+                            listener.scriptUpdated();
+                        }
+                    }
+                }
+            }
+        });
+    }
+
+    @Override
+    public ScriptType getScriptType() {
+        return ScriptType.GROOVY;
+    }
+
+    @Override
+    public String readScript() {
+        try (InputStream is = storage.readBinaryData(info.getId(), SCRIPT_CONTENT)
+                                     .orElseThrow(AssertionError::new)) {
+            return new String(ByteStreams.toByteArray(is), StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    @Override
+    public void writeScript(String content) {
+        try (Reader reader = new StringReader(content);
+             Writer writer = new OutputStreamWriter(storage.writeBinaryData(info.getId(), SCRIPT_CONTENT), StandardCharsets.UTF_8)) {
+            CharStreams.copy(reader, writer);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+        storage.updateModificationTime(info.getId());
+        storage.flush();
+        notifyDependencyListeners();
+    }
+
+    @Override
+    public void addListener(ScriptListener listener) {
+        Objects.requireNonNull(listener);
+        listeners.add(listener);
+    }
+
+    @Override
+    public void removeListener(ScriptListener listener) {
+        Objects.requireNonNull(listener);
+        listeners.remove(listener);
+    }
+
+    public ActionDb load(Network network) {
+        Objects.requireNonNull(network);
+        return new ActionDslLoader(readScript()).load(network);
```

Comment:
If I remember well, this method throws if a contingency element is not found. Maybe we should adapt this method to skip such contingencies

Reply:
Done, exceptions replaced by warnings